services/src/main/scala/cromwell/services/healthmonitor/impl/workbench/WorkbenchHealthMonitorServiceActor.scala

```diff
@@ -0,0 +1,71 @@
+package cromwell.services.healthmonitor.impl.workbench
+
+import java.net.URL
+
+import cats.data.Validated.{Invalid, Valid}
+import com.typesafe.config.Config
+import cromwell.cloudSupport.gcp.GoogleConfiguration
+import cromwell.cloudSupport.gcp.gcs.GcsStorage
+import cromwell.cloudSupport.gcp.genomics.GenomicsFactory
+import cromwell.core.WorkflowOptions
+import cromwell.services.healthmonitor.HealthMonitorServiceActor
+import cromwell.services.healthmonitor.HealthMonitorServiceActor.{MonitoredSubsystem, OkStatus, SubsystemStatus}
+import cromwell.services.healthmonitor.impl.CommonMonitoredSubsystems
+import net.ceedubs.ficus.Ficus._
+
+import scala.concurrent.Future
+
+/**
+  * A health monitor implementation which will monitor external services used in the Workbench ecosystem, such
+  * as GCS and PAPI
+  */
+class WorkbenchHealthMonitorServiceActor(val serviceConfig: Config, globalConfig: Config)
+  extends HealthMonitorServiceActor
+    with CommonMonitoredSubsystems {
+  implicit val as = context.system
+
+  override lazy val subsystems: List[MonitoredSubsystem] = List(DockerHub, EngineDb, Papi, Gcs)
+
+  private lazy val Gcs = MonitoredSubsystem("GCS", checkGcs _)
+  private lazy val Papi = MonitoredSubsystem("PAPI", checkPapi _)
+
+  val googleConfig = GoogleConfiguration(globalConfig)
+  val papiBackendName = serviceConfig.as[Option[String]]("services.HealthMonitor.PapiBackendName").getOrElse("Jes")
+  val papiConfig: Config = globalConfig.as[Config]("backend.providers.Jes.config")
+
+  /**
+    * Demonstrates connectivity to GCS by stat-ing the root execution bucket for Cromwell
+    */
+  private def checkGcs(): Future[SubsystemStatus] = {
+    val gcsAuthName = papiConfig.as[String]("filesystems.gcs.auth")
+    val rootExecutionBucket = papiConfig.as[String]("root").replace("gs://", "")
+
+    val cred = googleConfig.auth(gcsAuthName) match {
+      case Valid(a) => a.credential(WorkflowOptions.empty)
```

Comment:
In a firecloud configuration I'm not sure this would work. Their `filesystems.gcs.auth` is going to be a refresh token mode which you'll need a refresh token from the workflow options for.
Unless we specifically want to test authentication I think stat-ing a public bucket would give (almost ?) as much confidence in our ability to talk to GCS ?

Reply:
Discussed in person, will ping public bucket